spring-boot-cli/src/test/java/org/springframework/boot/cli/SampleIntegrationTests.java

```diff
@@ -32,112 +38,204 @@
  * @author Greg Turnquist
  * @author Roy Clarkson
  * @author Phillip Webb
+ * @author Tomek Kopczynski
  */
 public class SampleIntegrationTests {
 
-	@Rule
-	public CliTester cli = new CliTester("samples/");
+	private static final int PORT = 9000;
+
+	private CommandLineInvoker invoker = new CommandLineInvoker();
+	private CommandLineInvoker.Invocation invocation = null;
+
+	@After
+	public void clean() throws IOException {
+		if (this.invocation != null) {
+			if (this.invocation.isAlive()) {
+				// kill daemon java process
+				URI.create(String.format("http://localhost:%d/shutdown", PORT)).toURL().openStream().close(); // don't care about the output
+			}
+			this.invocation.destroy();
+		}
+		this.invocation = null;
+	}
+
+	private String callEndpoint(String uri) {
+		try {
+			InputStream stream = URI.create("http://localhost:" + PORT + uri).toURL()
+					.openStream();
+			BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+			String line;
+			StringBuilder result = new StringBuilder();
+
+			while ((line = reader.readLine()) != null) {
+				result.append(line);
+			}
+
+			return result.toString();
+		}
+		catch (Exception ex) {
+			throw new IllegalStateException(ex);
+		}
+	}
 
 	@Test
 	public void appSample() throws Exception {
-		String output = this.cli.run("app.groovy");
+		this.invocation = this.invoker.invoke("run", "samples/app.groovy");
 		URI scriptUri = new File("samples/app.groovy").toURI();
-		assertThat(output).contains("Hello World! From " + scriptUri);
+
+		Assert.assertThat(this.invocation.await(), Matchers.equalTo(0));
+		Assert.assertThat(this.invocation.getErrorOutput().length(), Matchers.equalTo(0));
+		Assert.assertThat(this.invocation.getOutput(), Matchers.containsString("Hello World! From " + scriptUri));
```

Comment:
This is applicable to all the tests that were changed. This made the body of the significantly more complex. Can you please remove those fully qualified call for a start and abstract the await part so that it doesn't have to ve repeated in all tests?